### PR TITLE
remove perf predictor flag

### PR DESCRIPTION
**Pull request by bridiver**
Resolves https://github.com/brave/brave-browser/issues/17536
Resolves https://github.com/brave/brave-browser/issues/10647

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_perf_predictor/browser/BUILD.gn [hunk 1]
```diff
@@ -1,17 +1,4 @@
...
-import("//build/buildflag_header.gni")
-import("buildflags/buildflags.gni")
-
-buildflag_header("buildflags") {
-  header = "buildflags.h"
-  flags = [ "ENABLE_BRAVE_PERF_PREDICTOR=$enable_brave_perf_predictor" ]
-}
-
 static_library("browser") {
-  # Remove when https://github.com/brave/brave-browser/issues/10647 is resolved
```
Comment (bsclifton) [inline components/brave_perf_predictor/browser/BUILD.gn:11]: I didn't run `npm run gn_check` locally - does this PR have any violations after this change?
Can we also close out https://github.com/brave/brave-browser/issues/10647?

**Comment by bsclifton** [comment]
@bridiver is there an issue for this?

Also curious (comment left) is https://github.com/brave/brave-browser/issues/10647 resolved?

File: components/brave_perf_predictor/browser/BUILD.gn (hunk 1 above)
Comment (bridiver) [reply components/brave_perf_predictor/browser/BUILD.gn:11]: not when I ran locally

Comment (bridiver) [reply components/brave_perf_predictor/browser/BUILD.gn:11]: I'll include that issue in `fixes` along with the issue I create for this

**Review by bsclifton — APPROVED** [review]
Needs rebase - but it's like someone is cutting onions in here. This is so beautiful that it's bringing tears to my eyes. A++ 😻

File: test/BUILD.gn [hunk 2]
```diff
@@ -215,6 +218,7 @@ test("brave_unit_tests") {
     "//components/domain_reliability:domain_reliability",
     "//components/language/core/browser",
     "//components/os_crypt:test_support",
+    "//components/page_load_metrics/common",
```
Comment (goodov) [inline test/BUILD.gn:221]: shouldn't we add ```//components/page_load_metrics/common:page_load_metrics_mojom``` here?
```bandwidth_savings_predictor_unittest.cc``` includes ```components/page_load_metrics/common/page_load_metrics.mojom.h```

File: components/brave_perf_predictor/browser/BUILD.gn (hunk 1 above)
Comment (iefremov) [reply components/brave_perf_predictor/browser/BUILD.gn:11]: But we still have `    deps += [ "//brave/browser/android:android_browser_process" ]` below that should be fixed

**Review by iefremov — APPROVED** [review]

File: test/BUILD.gn (hunk 2 above)
Comment (bridiver) [reply test/BUILD.gn:221]: shouldn't we get a gn check error if that was the case?

File: components/brave_perf_predictor/browser/BUILD.gn (hunk 1 above)
Comment (bridiver) [reply components/brave_perf_predictor/browser/BUILD.gn:11]: @iefremov that was necessary because of the existing override in DEPS

Comment (bridiver) [reply components/brave_perf_predictor/browser/BUILD.gn:11]: and yes, it should be fixed like other overrides, but isn't really in the scope of this PR

File: test/BUILD.gn (hunk 2 above)
Comment (goodov) [reply test/BUILD.gn:229]: yeah, I was curious why it didn't error. it's actually included via ```//chrome/common```:
```
//brave/test:brave_unit_tests --[public]-->       
//brave/test:brave_test_support_unit --[public]-->
//chrome/common:common --[public]-->
//components/page_load_metrics/common:page_load_metrics_mojom
```

File: chromium_src/chrome/browser/page_load_metrics/page_load_metrics_initialize.cc [hunk 3]
```diff
@@ -8,11 +8,7 @@
 #include "../../../../../chrome/browser/page_load_metrics/page_load_metrics_initialize.cc"
 #undef InitializePageLoadMetricsForWebContents
 
-#include "brave/components/brave_perf_predictor/browser/buildflags.h"
-
-#if BUILDFLAG(ENABLE_BRAVE_PERF_PREDICTOR)
 #include "brave/components/brave_perf_predictor/browser/perf_predictor_page_metrics_observer.h"
```
Comment (goodov) [inline chromium_src/chrome/browser/page_load_metrics/page_load_metrics_initialize.cc:11]: this include now can be moved to the top of the file

**Review by goodov — APPROVED** [review]

File: components/brave_perf_predictor/browser/BUILD.gn (hunk 1 above)
Comment (iefremov) [reply components/brave_perf_predictor/browser/BUILD.gn:11]: I'm saying that to get this fixed we need an issue on github

**Pull request merged**